go-controller/pkg/libovsdbops/model_client.go

```diff
@@ -386,52 +387,58 @@ func (m *modelClient) Lookup(opModels ...operationModel) error {
 }
 
 // lookup the model in the cache prioritizing provided indexes over a
-// predicate unless bulkop is set
+// predicate
 func (m *modelClient) lookup(opModel *operationModel) error {
 	if opModel.ExistingResult == nil && opModel.Model != nil {
 		opModel.ExistingResult = getListFromModel(opModel.Model)
 	}
 
 	var err error
-	if opModel.Model != nil && !opModel.BulkOp {
-		err = m.get(opModel)
-		if err == nil || err != client.ErrNotFound {
+	if opModel.Model != nil {
+		err = m.list(opModel)
+		if err == nil && reflect.Indirect(reflect.ValueOf(opModel.ExistingResult)).Len() == 0 {
+			// index was provided, but nothing was found
+			return client.ErrNotFound
+		}
```

Comment:
I tried to copy `get` behaviour when index is provided. `Get` will return `ErrNotFound` in that case, but I since it is ignored in `buildOps` https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/libovsdbops/model_client.go#L277 we can not return it here